cpp/oneapi/dal/algo/knn/test/badarg.cpp

```diff
@@ -0,0 +1,212 @@
+    REQUIRE_NOTHROW(this->train(knn_desc, this->get_train_data(), this->get_train_labels()));
+}
+
+KNN_BADARG_TEST("throws if both train data and labels are empty") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    REQUIRE_THROWS_AS(this->train(knn_desc, homogen_table{}, homogen_table{}), domain_error);
+}
+
+KNN_BADARG_TEST("throws if train labels are empty") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    REQUIRE_THROWS_AS(this->train(knn_desc, this->get_train_data(), homogen_table{}), domain_error);
+}
+
+KNN_BADARG_TEST("throws if train data is empty") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    REQUIRE_THROWS_AS(this->train(knn_desc, homogen_table{}, this->get_train_labels()), domain_error);
+}
+
+KNN_BADARG_TEST("throws if the number of train samples is not equal to the number of train labels") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    REQUIRE_THROWS_AS(this->train(knn_desc, this->get_train_data(3), this->get_train_labels()), domain_error);
+    REQUIRE_THROWS_AS(this->train(knn_desc, this->get_train_data(), this->get_train_labels(3)), domain_error);
+}
+
+KNN_BADARG_TEST("throws if the number of columns in labels is greater then one") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    REQUIRE_THROWS_AS(this->train(knn_desc, this->get_train_data(3), this->get_train_labels(3, 2)), domain_error);
+}
+
+KNN_BADARG_TEST("accept if infer data has suitable shape and not zero") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    const auto train_result = this->train(knn_desc, this->get_train_data(), this->get_train_labels());
+    const auto infer_data = this->get_infer_data();
+    REQUIRE_NOTHROW(this->infer(knn_desc, infer_data, train_result.get_model()));
+}
+
+/*KNN_BADARG_TEST("throws if the number of infer data columns exceeds the number of columns in train") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    const auto train_result = this->train(knn_desc, this->get_train_data(), this->get_train_labels());
+    const auto infer_data = this->get_infer_data(2, 3);
+    REQUIRE_THROWS_AS(this->infer(knn_desc, infer_data, train_result.get_model()), invalid_argument);
+}
+
+KNN_BADARG_TEST("throws if the number of infer data columns is less then the number of columns in train") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    const auto train_result = this->train(knn_desc, this->get_train_data(), this->get_train_labels());
+    const auto infer_data = this->get_infer_data(7, 1);
+    REQUIRE_THROWS_AS(this->infer(knn_desc, infer_data, train_result.get_model()), invalid_argument);
+}*/
```

Comment:
Add comment/TOOD to explain why it's commented and when it should be commented out. For the person who don't know the reason it looks like trash code that somebody forgot to remove.

Reply:
Done